array_api_compat/common/_helpers.py

```diff
@@ -956,7 +951,7 @@ def is_writeable_array(x: object) -> bool:
     return is_array_api_obj(x)
 
 
-def is_lazy_array(x: object) -> bool:
+def is_lazy_array(x: object) -> TypeGuard[_ArrayApiObj]:
```

Comment:
Anything with a `shape: tuple` that contains a `None` would return `True` here, so this isn't correct.

Reply:
That's an issue with the implementation, not the type annotation.
As per the docstring, this function returns True if and only if x is an array API object and it is lazy.